src/core/ObservableQuery.ts

```diff
@@ -776,7 +776,10 @@ Did you mean to call refetch(variables) instead of refetch({ variables })?`,
 
     const maybeFetch = () => {
       if (this.pollingInfo) {
-        if (!isNetworkRequestInFlight(this.queryInfo.networkStatus)) {
+        if (
+          !isNetworkRequestInFlight(this.queryInfo.networkStatus) &&
+          (!this.options.focusPolling || document?.hasFocus())
```

Comment:
```suggestion
          (!this.options.focusPolling || (typeof document !== 'undefined' && document.hasFocus()))
```

`document` doesn't exist in Node.js server environments, so you'll want to guard against that here to ensure its a defined value in the global scope before checking `hasFocus` on it.

Reply:
Optional chaining used in `document?.hasFocus()` should serve the same purpose as `(typeof document !== 'undefined' && document.hasFocus())`

Reply:
Not quite. Using `document` in this manner assumes that `document` is available in the global scope. If `document` isn't available in global scope, you'll get a `ReferenceError`.

`typeof document` will first check to see if that variable is available in global scope, so its safe to execute even if you're working in a non-browser environment.

To try this yourself, open up a Node repl and type `document`

```sh
$ node
Welcome to Node.js v18.17.1.
Type ".help" for more information.
> document
Uncaught ReferenceError: document is not defined 
```

Notice the `Uncaught ReferenceError` here.

Now if you try `typeof document`, you'll note it doesn't crash, but returns `undefined`

```sh
> typeof document
'undefined'
```

Reply:
Ah this makes sense. Thanks for letting me know about it.